### ISSUE TITLE
setting future date for backend recurring contribution ignored

### ISSUE THREAD

**Issue by lcdservices** [iATSPayments/com.iatspayments.civicrm#197]
If you create a recurring credit card contribution via the backend and select a start date in the future, the setting is ignored -- it processes the payment immediately. Is this a supported feature for iATS? If not, we should suppress the fields to avoid confusion.

**Comment by adixon** [comment]
Good point - the ability to set a future start date for recurring contributions is a relatively new feature in core that we haven't looked at, so I think you are correct that it won't work as advertised.

I suspect it won't be difficult to implement, we're already dealing with future start dates with the fixed recurring dates features.

I'll add a note to the current release notes and add this to our todo for the next release.

**Comment by lcdservices** [comment]
that would be great. happy to do some testing when you get a beta version done.

**Comment by lcdservices** [comment]
@adixon just curious if you have a general time frame for the next release. thx.

**Comment by adixon** [comment]
Well, let's hope it's a lot less than the last one! I think within a month or two is likely - given that it was a first 4.7 release, there are likely to be small unexpected blips.

**Comment by adixon** [comment]
Interestingly - this might be a core issue. The date field that lets you set a future start date for recurring contributions should only be exposed if the payment processor specifically implements the appropriate method (supportsRecurringFutureStartDates), and this processor doesn't.

**Comment by lcdservices** [comment]
I noticed that as well. Though other payment processors seem to handle it properly (A.net enables it, PayPal disables, etc.).

**Comment by adixon** [comment]
With which versions of CiviCRM and iATS Payments are you seeing this?

**Comment by lcdservices** [comment]
Civi 4.7.19 and iATS 1.6.0

**Comment by lcdservices** [comment]
@adixon just wondering if you still think this will be handled in the next minor release, and if so, are you at a point where you have a ballpark for when that might be released (or in beta stage for testing)

**Comment by adixon** [comment]
Thanks for asking - do you want to try out this? https://github.com/iATSPayments/com.iatspayments.civicrm/tree/iATS-managed-recurring
It includes support for the CiviCRM core future start date option for both cc and ach/eft and also the option to allow that for public forms (in the iats config page, there's a new checkbox).

**Comment by lcdservices** [comment]
I implemented that branch and ran some tests and it didn't work.
- created a credit card contribution from the backend for tomorrow, and the transaction was processed immediately
- visited the iATS settings page but did not see the frontend future date option you referred to. I saw several other new setting options, but not that one.

**Comment by adixon** [comment]
I just ran through a few tests on a test-upgraded-client site and it's working as designed. I suspect you were looking at master or the other branch I've been working on?

**Comment by lcdservices** [comment]
I pulled the latest master branch (beta-1) and ran through more tests. I now see the component setting option to enable setting a future date through the frontend form. I ran a test and the contribution was created with the selected future receive date -- so that appears to be working (I will know for certain once it processes). However, multiple tests with the backend form still are not successful -- in every case the contribution is still dated today.
